pkgs/top-level/perl-packages.nix

```diff
@@ -20404,7 +20404,8 @@ let
       url = mirror://cpan/authors/id/T/TO/TODDR/XML-Parser-2.46.tar.gz;
       sha256 = "0pai3ik47q7rgnix9644c673fwydz52gqkxr9kxwq765j4j36cfk";
     };
-    patchPhase = stdenv.lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    patches = [ ../development/perl-modules/xml-parser-0001-HACK-Assumes-Expat-paths-are-good.patch ];
```

Comment:
maybe
```
patches = stdenv.lib.optional (stdenv.buildPlatform != stdenv.hostPlatform)
 ../development/perl-modules/xml-parser-0001-HACK-Assumes-Expat-paths-are-good.patch;
```

Reply:
When the patch has no side-effect in actual use, I prefer not conditionally applying them.

One reason is that when it fails to apply in the future, it will not be seen unless tested in that particular condition.

Though, one improvement that could be done is to always apply the patch, and then apply the conditional depending on a parameter to `perl Makefile.PL` which I tried to do, but I'm not a perl developer and couldn't manage to make it take the parameter like `EXPATLIBPATH` but `ASSUME_EXPAT_IS_VALID` or something of the sort. With such a change, we would add `ASSUME_EXPAT_IS_VALID` only when the validations fail.